assets/activation-scripts/activate

```diff
@@ -15,6 +15,69 @@ export _tcsh_home="@out@/activate.d/tcsh_home"
 export _flox_activate_tracelevel="${_FLOX_PKGDB_VERBOSITY:-0}"
 [ "$_flox_activate_tracelevel" -eq 0 ] || set -x
 
+NOT_READY="SOCKET_NOT_READY"
+
+poll_services_status () {
+  local socket_file="$1"
+  local output
+  output=$(process-compose process list -u "$socket_file" -o json 2>&1)
+  # Save whatever the current `pipefail` behavior is so it can be restored
+  local saved_options
+  saved_options=$(set +o)
+  # We don't want to exit on pipe failures here
+  set +o pipefail
+  local parsed_json
+  parsed_json=$(echo "$output" | jq -r -c '.[0].status' 2>/dev/null)
+  # Restore the previous shell settings
+  eval "$saved_options"
+  # `parsed_json` will be a null string if anything went wrong
+  echo "${parsed_json:-${NOT_READY}}"
+}
+
+wait_for_services_socket () {
+  local socket_file="$1"
+  local status
+  status=$(poll_services_status "$socket_file")
+  while [ "$status" == "$NOT_READY" ]; do
+    sleep 0.1
+    status=$(poll_services_status "$socket_file")
+  done
+}
+
+start_services_blocking () {
+  local config_file="$1"; shift
+  local socket_file="$1";
+  local log_file
+  log_file="${_FLOX_SERVICES_LOG_FILE:-$(mktemp)}"
+  # process-compose will vomit all over your log files unless you tell it otherwise
+  local previous_no_color="${NO_COLOR:-}"
+  export NO_COLOR=1
```

Comment:
(non-blocking, polish later) Looks like we could later move this into the rendered config which would be more explicit and require less shell juggling: https://f1bonacc1.github.io/process-compose/logging/#logger-configuration